engine/common/synchronization/config.go

```diff
@@ -68,10 +73,16 @@ type SpamDetectionConfig struct {
 }
 
 func NewSpamDetectionConfig() *SpamDetectionConfig {
```

Comment:
Better to return an error and let the parent context throw it so that the node shuts down properly

Reply:
91607b899ecb1dbc4414582de6c27817a388c8f6